### PR TITLE
Updating test structure

### PR DESCRIPTION
**Pull request by shrutivanw**
Updating tests to have nested describes and correct spacing. Updating readme file.
Note 1: The solution branch will be updated to have the same changes.
Note 2: all other CS Fundamentals' homework will be following this format.

### CODE REVIEW

File: specs/string_reverse_spec.rb [hunk 1]
```diff
@@ -2,53 +2,70 @@
...
-    string_reverse(test_string)
-    test_string.must_equal "erawtfoS"
-  end
-end
+        string_reverse(test_string)
+
+        test_string.must_equal "olleH"
+      end
+    end
+    describe "test 2" do
```
Comment (kariabancroft) [inline specs/string_reverse_spec.rb]: I think you're using extra `describe` blocks here. Since this is really modeled off of a BDD framework, ideally everything that is a text block has some sort of explanation for the expected behavior. I think you could have the single `describe` block at the top-level for "basic tests" and then dive straight into the `it` blocks that have the appropriate behavior descriptions. (The "test 1" and "test 2" describe blocks are really what is concerning, so you could either add additional descriptions to these or collapse them down - whichever makes sense)

File: specs/string_reverse_spec.rb [hunk 2]
```diff
@@ -2,53 +2,70 @@
...
-end
 
-describe "edge case 2 - string reverse" do
-  it "a really long string passed to reverse" do
-    test_string = "Oh, Supercalifragelisticexpialidocious! Even though the sound of it is something quite atrocious. If you say it loud enough, you'll always sound precocious. Supercalifragelisticexpialidocious!"
-    string_reverse(test_string)
-    test_string.must_equal "!suoicodilaipxecitsilegarfilacrepuS .suoicocerp dnuos syawla ll'uoy ,hguone duol ti yas uoy fI .suoicorta etiuq gnihtemos si ti fo dnuos eht hguoht nevE !suoicodilaipxecitsilegarfilacrepuS ,hO"
+  # check for edge cases
+  describe "edge cases" do
+    describe "edge case 1" do
```
Comment (kariabancroft) [inline specs/string_reverse_spec.rb]: Same comment re: describe blocks and their descriptions

File: specs/string_reverse_spec.rb (hunk 1 above)
Comment (shrutivanw) [reply specs/string_reverse_spec.rb]: Sounds good! It's likely that I misunderstood Charles's feedback and am glad that you are reviewing it. Thank you.

**Comment by shrutivanw** [comment]
Thank you, @kariabancroft PR updated. Once we get to consensus for this PR, I'll apply the same fix to the other PRs.

File: README.md [hunk 3]
```diff
@@ -3,5 +3,7 @@ In this assignment, you'll design and implement one of the string manipulation f
 Remember that a string is an array of characters. Algorithms that worked on restricted arrays will work on strings as well.
 
 ## Exercise
-Design and implement a method to reverse a string *in place*. For example, if the method is called with input parameter of "Lovelace", the method should update the input string object to have the value "ecalevoL". Share and explain the time and space complexities for your solution.
-**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`. 
+* Design and implement a method to reverse a string *in place*. For example, if the method is called with input parameter of "Lovelace", the method should update the input string object to have the value "ecalevoL".
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
Comment (Hamled) [inline README.md]: Similar comment to one of @kariabancroft's comments on the `reverse_words` repo: It seems likely that we want an extra blank line above this so that the note ends up in its own paragraph.

**Review by Hamled — APPROVED** [review]
Looks good. I would consider moving the empty string and nil tests into the edge cases section, while moving the long sentence case into the basic tests section.

**Comment by shrutivanw** [comment]
Thank you @Hamled I moved the nil and empty string test cases into edge cases and added a newline so readme.md would render correctly.

I think the long string test case belongs in edge cases so I left it there - as is. We can chat further on that one if you feel strongly about it. :-)
